qt --with-docs build fails

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [X] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [X] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [X] ran `brew update` and can still reproduce the problem?
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

-------------

`brew install qt --with-docs` is failing for me, and [for @RandomDSDevel](https://github.com/Homebrew/homebrew-core/issues/28455#issuecomment-397417316). It complains that it cannot find `qdoc`, which (I think) should have been produced and installed as part of the build.

```
make[5]: Nothing to be done for `prepare_docs'.
cd corelib/ && ( test -e Makefile || /private/tmp/qt-20180625-30856-16o7mcc/qt-everywhere-src-5.11.1/qtbase/bin/qmake -o Makefile /private/tmp/qt-20180625-30856-16o7mcc/qt-everywhere-src-5.11.1/qtbase/src/corelib/corelib.pro ) && /Applications/Xcode-9.4.app/Contents/Developer/usr/bin/make -f Makefile prepare_docs
/usr/local/Cellar/qt/5.11.1/bin/qtattributionsscanner /private/tmp/qt-20180625-30856-16o7mcc/qt-everywhere-src-5.11.1/qtbase --filter QDocModule=qtcore -o /private/tmp/qt-20180625-30856-16o7mcc/qt-everywhere-src-5.11.1/qtbase/src/corelib/codeattributions.qdoc
/private/tmp/qt-20180625-30856-16o7mcc/qt-everywhere-src-5.11.1/qtbase/src/corelib/qdoc_wrapper.sh -outputdir /private/tmp/qt-20180625-30856-16o7mcc/qt-everywhere-src-5.11.1/qtbase/doc/qtcore -installdir /usr/local/Cellar/qt/5.11.1/doc /private/tmp/qt-20180625-30856-16o7mcc/qt-everywhere-src-5.11.1/qtbase/src/corelib/doc/qtcore.qdocconf -prepare -no-link-errors -I. -Iglobal -I../3rdparty/harfbuzz/src -I../3rdparty/md5 -I../3rdparty/md4 -I../3rdparty/sha3 -I../3rdparty/double-conversion/include -I../3rdparty/double-conversion/include/double-conversion -I../3rdparty/forkfd -I../../include -I../../include/QtCore -I../../include/QtCore/5.11.1 -I../../include/QtCore/5.11.1/QtCore -I.moc -I.tracegen -I../3rdparty/pcre2/src -I../../mkspecs/macx-clang
/private/tmp/qt-20180625-30856-16o7mcc/qt-everywhere-src-5.11.1/qtbase/src/corelib/qdoc_wrapper.sh: line 12: /usr/local/Cellar/qt/5.11.1/bin/qdoc: No such file or directory
/private/tmp/qt-20180625-30856-16o7mcc/qt-everywhere-src-5.11.1/qtbase/src/corelib/qdoc_wrapper.sh: line 12: exec: /usr/local/Cellar/qt/5.11.1/bin/qdoc: cannot execute: No such file or directory
make[5]: *** [prepare_docs] Error 126
make[4]: *** [sub-corelib-prepare_docs] Error 2
make[3]: *** [sub-src-prepare_docs] Error 2
make[2]: *** [module-qtbase-prepare_docs] Error 2
make[1]: *** [html_docs] Error 2
make: *** [docs] Error 2
```

Build logs: https://gist.github.com/e6698f570066bfe5f0b767678824518c

Looks like the `qdoc` tool either isn't getting built, or isn't getting installed, so the "make docs" step is failing.

Trying to narrow down the cause; it's a big build to go through.

> Is it possible that the issue is that qdoc hasn't been installed into the prefix yet and/or isn't in the PATH?

That appears to be the issue. It doesn't look like qdoc is getting built at all.

```
make[5]: *** [prepare_docs] Error 126
make[4]: *** [sub-corelib-prepare_docs] Error 2
make[3]: *** [sub-src-prepare_docs] Error 2
make[2]: *** [module-qtbase-prepare_docs] Error 2
make[1]: *** [html_docs] Error 2
make: *** [docs] Error 2
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: make docs
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 5
When you exit this shell, you will return to the menu.
angharad% find . -name qdoc -type f
angharad% find $(/usr/local/bin/brew --prefix qt) -name qdoc
angharad%
```

I'm looking through the Makefiles etc to see if it needs to be explicitly invoked or something.

> ... I haven't tested Qt v5.11.1 yet ...

I tried building 5.11.1; happens there too.

--- a/Formula/qt.rb
+++ b/Formula/qt.rb
@@ -17,7 +17,6 @@ class Qt < Formula
 
   keg_only "Qt 5 has CMake issues when linked"
 
-  option "with-docs", "Build documentation"
   option "with-examples", "Build examples"
   option "without-proprietary-codecs", "Don't build with proprietary codecs (e.g. mp3)"
 
@@ -79,11 +78,6 @@ class Qt < Formula
     system "make"
     ENV.deparallelize
     system "make", "install"
-
-    if build.with? "docs"
-      system "make", "docs"
-      system "make", "install_docs"
-    end
 
     # Some config scripts will only find Qt in a "Frameworks" folder
     frameworks.install_symlink Dir["#{lib}/*.framework"]

